optuna/trial/_trial.py

```diff
@@ -59,6 +62,7 @@ def __init__(self, study: "optuna.study.Study", trial_id: int) -> None:
             study, self._cached_frozen_trial
         )
         self._relative_params: Optional[Dict[str, Any]] = None
+        self._fixed_params = self._cached_frozen_trial.system_attrs.get("fixed_params", [])
```

Comment:
```suggestion
        self._fixed_params = self._cached_frozen_trial.system_attrs.get("fixed_params", {})
```
This would be better as a dict, since `fixed_params` type is `dict`.

Reply:
Thank you. It makes sense.